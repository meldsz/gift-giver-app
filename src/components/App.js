import React, { Component } from 'react';
import { Button } from "react-bootstrap";
import Gift from "./Gift";
import { max_number } from "./helper";

class App extends Component {
    constructor() {
        super();
        this.state = {
            gifts: []
        }
    }

    addGift = () => {
        const { gifts } = this.state;
        gifts.push({ id: max_number(gifts.map(gift => gift.id)) + 1 });
        this.setState({ gifts });
    }

    removeGift = (id) => {
        const gifts = this.state.gifts.filter(gift => gift.id !== id);
        this.setState({ gifts });
    }

    render() {
        return (
            <div>
                <h2>Gift Giver</h2>
                <div className="gift-list">
                    {this.state.gifts.map(gift =>
                        <Gift key={gift.id} gift={gift} removeGift={(id) => this.removeGift(id)} />
                    )}
                </div>
                <Button onClick={this.addGift}>Add Gift</Button>
            </div>
        )
    }
}

export default App;