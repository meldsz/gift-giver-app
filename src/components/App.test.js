import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe('App', () => {
    const app = shallow(<App />);
    it('app renders correctly', () => {
        expect(app).toMatchSnapshot();
    });

    it('sets the `state` to empty array of gifts', () => {
        expect(app.state().gifts).toEqual([]);
    });

    describe('when clicking on the add gift button', () => {

        beforeEach(() => {
            app.find('Button').simulate('click');
        });

        afterEach(() => {
            app.setState({ gifts: [] });
        });

        it('adds a gift to the `state` when the add gift button is clicked', () => {
            expect(app.state().gifts).toEqual([{ id: 1 }]);
        });

        it('renders gifts on the render on clicking add gift button', () => {
            expect(app.find('.gift-list').children().length).toEqual(1);
        });

        it('renders Gift', () => {
            expect(app.find('Gift').exists()).toBe(true);
        });
    });

    


});