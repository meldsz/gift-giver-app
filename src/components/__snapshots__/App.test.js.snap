// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App app renders correctly 1`] = `ShallowWrapper {}`;
